sklearn/datasets/tests/test_base.py

```diff
@@ -391,3 +396,263 @@ def test_fetch_remote_raise_warnings_with_invalid_url(monkeypatch):
         for r in record:
             assert str(r.message) == f"Retry downloading from url: {url}"
         assert len(record) == 3
+
+
+def test_derive_folder_and_filename_from_url():
+    folder, filename = _derive_folder_and_filename_from_url(
+        "https://example.com/file.tar.gz"
+    )
+    assert folder == "example.com"
+    assert filename == "file.tar.gz"
+
+    folder, filename = _derive_folder_and_filename_from_url(
+        "https://example.com/نمونه نماینده.data"
```

Comment:
Apparently this means representative sample for those who wonder :wink:

Reply:
It's so meta, like the original contents of the corrupted file ;)